server/dns_server.ml

```diff
@@ -686,10 +719,14 @@ let update_data trie zone (prereq, update) =
   else
     let zones =
       (* accumulate old and new zones *)
-      let one t s =
-        Dns_trie.fold Rr_map.Soa t (fun z _ s -> Domain_name.Set.add z s) s
-      in
-      one trie' (one trie Domain_name.Set.empty)
+      Domain_name.Map.fold (fun name _ acc ->
+          match Dns_trie.zone name trie', Dns_trie.zone name trie with
+          | Ok (z, _), _ -> Domain_name.Set.add z acc
+          | _, Ok (z, _) -> Domain_name.Set.add z acc
```

Comment:
1) I don't really understand what's going on here, but in line 733 there's:
    ```ocaml
          | Ok oldsoa, Ok soa when Soa.newer ~old:oldsoa soa ->
  The `z`from `trie` doesn't get added when `Dns_trie.zone name trie'` works, tough, so the `oldsoa` will presumably come from `trie'` - will that be a problem, or are they guaranteed to be identical or something like that in the `Ok oldsoa, Ok soa` case?

2) It would be nice if there was consistency in the matches so `trie'` is always before/after `trie`, instead of random ordering. It's hard enough to spot the `'` in the first place. :)